react-components/src/components/Reveal3DResources/Reveal3DResources.tsx

```diff
@@ -54,23 +52,33 @@ export const Reveal3DResources = ({
   image360Settings
 }: Reveal3DResourcesProps): ReactElement => {
   const viewer = useReveal();
-
   const [reveal3DModels, setReveal3DModels] = useState<TypedReveal3DModel[]>([]);
-
   const numModelsLoaded = useRef(0);
 
+  useRemoveNonReferencedModels(resources, viewer);
+
+  const { classicModelOptions, cadOrPointCloudResources } = useCadOrPointCloudResources(resources);
+
   useEffect(() => {
-    void getTypedModels(resources, viewer, onResourceLoadError).then(setReveal3DModels);
-  }, [resources, viewer]);
+    const fetchTypedModels = async (): Promise<void> => {
+      const typedModels = await getTypedModels(
+        classicModelOptions,
+        viewer,
+        cadOrPointCloudResources,
+        onResourceLoadError
+      );
+      setReveal3DModels(typedModels);
+    };
+
+    void fetchTypedModels();
```

Comment:
I know it's not your code initially, but it does smell like it should be a `useQuery` :eyes: